runtime/src/main/java/org/corfudb/runtime/BootstrapUtil.java

```diff
@@ -35,12 +37,51 @@
      * @param retries      Number of retries to bootstrap each node before giving up.
      * @param retryTimeout Duration between retries.
      */
-    public static void bootstrap(@Nonnull Layout layout,
+    public static void bootstrap(@NonNull Layout layout,
                                  int retries,
-                                 @Nonnull Duration retryTimeout) {
+                                 @NonNull Duration retryTimeout) {
         bootstrap(layout, CorfuRuntimeParameters.builder().build(), retries, retryTimeout);
     }
 
+    /**
+     * Bootstraps a layout server connected to the specified router.
+     *
+     * @param router Router connected to the node.
+     * @param layout Layout to bootstrap with
+     */
+    private static void bootsrapLayoutServer(IClientRouter router, Layout layout)
```

Comment:
I think you can factor out bootsrapLayoutServer and bootsrapManagementServer

Reply:
The completion criteria for both these methods are different. The try catch blocks of both cannot be combined. When we get an AlreadyBootstrappedException on bootstrapping the layout server, we would still want to continue the process and go ahead with bootstrapping the management server. So they cannot be combined.
Putting both these smaller methods in the main method increases the code cognitive complexity reported by sonarqube.